components/admin/lessons/AdminLessonInfo.tsx

```diff
@@ -152,7 +152,7 @@ export const AdminLessonInfo: React.FC<LessonInfoProps> = ({
   selectedLesson
 }) => {
   // true when user clicks on `create new lesson` button
-  if (lessons && selectedLesson === lessons.length - 1) {
+  if (lessons && selectedLesson === lessons.length) {
```

Comment:
There is lessons useState value at lessons page. It's passed to AdminLessonsSideBar where it is cloned and `Create new lesson` title is appended. Lessons themselves are not modified. So if you render 3 lessons, AdminLessonsSideBar will have 4 buttons and etc.